### DISCUSSION TITLE
HyperSerialWLED vs HyperSerialESP32 Performance

### DISCUSSION THREAD

**Discussion by AZDane** [awawa-dev/HyperHDR#649] [General]
I initially Flashed my ESP32 with HyperSerialWLED to reap the benefits of using WLED, but eventually had to go to HyperSerialESP32 because of flickering issues. Are there any chance a future update will eliminate these issues or is it just too much for the ESP32 to handle? I tried with the lower speed version, but still encountered the same issues.

**Comment by awawa-dev** [comment]
Don't know if it's fixed in upstream.  Did you try to use latest WLED 0.14x (testing various speeds with AWA option disabled in HyperHDR).

**Reply by AZDane** [reply]
I did try using WLED 14.0 b6 and disabling the AWA option, but my results were poor. 
Basically I had to stick with the lowest baud rate or the ESP32 would basically freeze up and become unresponsive if exposed to high speeds. 
My ESP32 uses CP2104.

**Reply by awawa-dev** [reply]
I don't know about your case but if you use Neopixel LED strip (sk6812/ws2812b and others) then as far as I know WLED uses the RMT method to render LEDs. RMT has a side effect of massively using the IRQ to replenish a tiny data buffer which is then used to render the LEDs. This may cause a conflict with other peripherals, e.g. the serial port or wifi, and vice versa: the serial port or wifi may disturb RMT rendering. You can find more in the Neopixelbus project used by WLED. In HyperSerialESP32 I use an alternative I2S rendering method without such side effects. Maybe you can reopen the issue on the WLED project if you think that your ESP is affected the same way? `https://github.com/Aircoookie/WLED/issues/2618#issuecomment-1738960132`

**Reply by AZDane** [reply]
Thank you, I will do that.

**Reply by advanced03** [reply]
Any advantages over using hyper wled over normal wled?

**Reply by AZDane** [reply]
HyperWLED is better optimized for Ambilight you will get lower latency and better stability. HyperWLED is basically stripped down for better performance.
Personally I had to eventually skip WLED functionality completely and go with HyperSerialESP32.
